### PR TITLE
Product models

### PR DESCRIPTION
**Pull request by siteations**
Core models of brewType, product, photo, review, unit... examine the associations because they seem hairy. Also... I've left the unique index elements on options but weeded-out other hooks and virtuals.

### CODE REVIEW

File: db/models/index.js [hunk 1]
```diff
@@ -18,28 +18,23 @@ const Photo = require('./photo');
...
-  // BrewID... unique brew types... each product can have one
-  // SellerID... one seller per product
-  // Photos... multiple to one product entry
-
 OAuth.belongsTo(User);
 User.hasOne(OAuth);
 
-Product.belongsTo(Seller); //sellerID on product
-Seller.hasMany(Product); //symmetrical
+Product.belongsTo(Seller, {as: 'seller'}); //sellerID on product
```
Comment (ayoung09) [inline db/models/index.js]: I believe if the alias is the same as the table name, it is redundant (without it, sellerId would still be on Product)

File: db/models/index.js [hunk 2]
```diff
@@ -18,28 +18,23 @@ const Photo = require('./photo');
...
+Product.belongsTo(Unit, {as: 'unit'}); //unitID on product
 
-Product.hasMany(Photo); //productID on photo
+Product.hasMany(Photo, {as: 'product'}); //productID on photo
 
 //reviews - created from/after product by logged-in user
-Review.belongsTo(Product); //productID on review
-Review.belongsTo(User); //userID on review
+Review.belongsTo(Product, {as: 'product'}); //productID on review
+Review.belongsTo(User, {as: 'user'}); //userID on review
```
Comment (ayoung09) [inline db/models/index.js]: Question: should we enable photo uploads for reviews (and create an association between photo and review)?

File: db/models/brewType.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+const Sequelize = require('sequelize');
+const db = require('APP/db');
+
+const BrewType = db.define('brews', {
+  name: {
+    type: Sequelize.STRING,
+    allowNull: false,
+    validate: {
+
+    }
```
Comment (ayoung09) [inline db/models/brewType.js]: should there be a validation here?

File: db/models/brewType.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  },
+  flavor: {
+    type: Sequelize.ARRAY(Sequelize.TEXT),
```
Comment (ayoung09) [inline db/models/brewType.js]: should flavor only be on product?

**Review by ayoung09 — COMMENTED** [review]
A couple of comments on BrewType and Index

File: db/models/brewType.js (hunk 4 above)
Comment (siteations) [reply db/models/brewType.js]: keeping on both to search product and general brewTypes... recommendations on specific brands and general types.

File: db/models/brewType.js (hunk 3 above)
Comment (siteations) [reply db/models/brewType.js]: erased!

File: db/models/index.js (hunk 2 above)
Comment (siteations) [reply db/models/index.js]: yes! set-up as Review.hasMany(Photos.....

File: db/models/index.js (hunk 1 above)
Comment (siteations) [reply db/models/index.js]: fixed the redundancy of aliases (before inclusion of cart / order).

**Pull request merged**